modules/xmpp/XmppConnection.js

```diff
@@ -361,30 +367,41 @@ export default class XmppConnection extends Listenable {
 
     /**
      * Starts the Websocket keep alive if enabled.
+     * @param {String} options.shard - The BOSH or WebSocket was connected to this shard.
+     * @param {Number} [options.websocketKeepAlive=240000] - The websocket keep alive interval.
+     * Pass 0 to immediately check and then proceed with the interval from options.
      *
      * @private
      * @returns {void}
      */
-    _maybeStartWSKeepAlive() {
-        const { websocketKeepAlive } = this._options;
+    _maybeStartWSKeepAlive(
+            { shard, websocketKeepAlive },
```

Comment:
passing the shard and options  to override behavior is confusing, I think you did that to be able to do the fetch and check on demand/instantly, but this should be a separate function (see the comment above)